app/controllers/change_answers_controller.rb

```diff
@@ -25,6 +27,23 @@ def update
 
 private
 
+  # While we're in a 'change answers loop', we want to be working with a temporary copy of the answers
+  # stored in a section of the session called 'pending'.
+  def session_data
+    data = super
+
+    # Set up a fresh copy of the answers into the pending section if it's blank or we're
+    # starting a new loop
+    if !data[:pending] || params[:begin_editing]
+      pending = data.dup
+      pending[:pending] = nil
```

Comment:
Question - So I'm guessing that CFE doesn't care that we are also sending it the extra key value pair `pending => nil`?

Reply:
Ah, but we're not: we don't send the session data straight to CFE. Instead, the CfeParamBuilders classes pick specific parts of the session data out and transform them into the format CFE wants. So the 'pending' key never gets near CFE.

Reply:
Oh yeah. Neat.